src/jaxsim/parsers/kinematic_graph.py

```diff
@@ -776,3 +881,30 @@ def pre_H_suc(
                 0
             ]
         )
+
+    def find_parent_link_of_frame(self, name: str) -> str:
+        """
+        Find the parent link of a frame.
+
+        Args:
+            name: The name of the frame.
+
+        Returns:
+            The name of the parent link of the frame.
+        """
+
+        try:
+            frame = self.graph.frames_dict[name]
+        except KeyError as e:
+            raise ValueError(f"Frame '{name}' not found in the kinematic graph") from e
+
+        match frame.parent.name:
+            case parent_name if parent_name in self.graph.links_dict:
+                return parent_name
+
+            case parent_name if parent_name in self.graph.frames_dict:
+                return self.find_parent_link_of_frame(name=parent_name)
+
+            case _:
+                msg = f"Failed to find element with name '{frame.parent.name}'"
```

Comment:
This looks a bit ambiguous to me, can we be a bit more specific?
```suggestion
                msg = f"Failed to find parent element of frame '{name}' with name '{frame.parent.name}'"
```